src/Compiler.php

```diff
@@ -252,15 +250,39 @@ protected function parseBindAttributes(string $attributeString): string
 
     protected function attributesToString(array $attributes): string
     {
-        return collect($attributes)
-            ->map(function ($value, string $attribute) {
-                if (is_array($value)) {
-                    $value = '['.$this->attributesToString($value).']';
-                }
+        $attributes = collect($attributes);
+
+        $string = [];
+
+        $string['plain'] = '['.$attributes
+                ->reject(function ($value, string $attribute) {
+                    return Str::startsWith($attribute, '...');
+                })
+                ->map(function ($value, string $attribute) {
+                    if (is_array($value)) {
+                        $value = $this->attributesToString($value);
+                    }
+
+                    return "'{$attribute}' => {$value}";
+                })
+                ->implode(',').']';
```

Comment:
Can we refactor this to do the collection mapping/implode first and then wrap the resulting string in square brackets? Instead of the string concatenation surrounding the mapping/implode

Reply:
sure, will do so

Reply:
https://github.com/spatie/laravel-blade-x/pull/112/commits/504601722c08859291832dd6f854c1daf8793689